Dev/11 add delete transportation

je-nnyt opened the pull request.

je-nnyt committed:
Create Transportation class

Add attributes, constructor, getters and setters.

je-nnyt committed:
Implement addTransport, deleteTransport and ArrayList<Transportation>

je-nnyt committed:
Refactor code

je-nnyt committed:
Change package name to adhere to coding standard

choiak commented on src/main/java/voyatrip/Trip.java:
```diff
@@ -0,0 +1,30 @@
+public class Trip {
+
+    private ArrayList<Transportation> transportations;
+    private Integer transportationCount;
```
I think there is unnecessary to store this Since ArrayList already can provide the number of item in the list.

je-nnyt replied:
Good point! I will remove it!

je-nnyt committed:
Merge branch 'master' into dev/11-add-delete-transportation

je-nnyt committed:
Remove transportationCount

The transportation count can be directly obtained from the size() method from the ArrayList

je-nnyt committed:
Remove transportationCount

The transportation count can be obtained directly from the size() method provided by the ArrayList

je-nnyt committed:
Fix compilation conflict

Missing bracket caused compilation issue. Add a break statement to stop searching for a transportation once the correct one is found.

je-nnyt committed:
Fix error provided by Gradle

Missing a newline at the end of file.

je-nnyt committed:
Fix newline error provided by Gradle

Missing a newline at the end of file.

je-nnyt committed:
Fix error provided by Gradle

Missing a newline at the end of file.

je-nnyt committed:
Fix newline error provided by Gradle

Missing a newline at the end of file.

je-nnyt committed:
Merge branch 'dev/11-add-delete-transportation' of https://github.com/je-nnyt/tp into dev/11-add-delete-transportation

# Conflicts:
#	src/main/java/voyatrip/Trip.java

je-nnyt committed:
Change parameter of deleteTransportation to type Integer

Deleting by index is more convenient for version 1.

je-nnyt committed:
Add a name attribute for Transportation class

Lyam-T reviewed (changes requested):
Please add back all the accidentally deleted files
<img width="322" alt="image" src="https://github.com/user-attachments/assets/75df561d-820a-44a3-93c9-a671ba043ac5" />

Lyam-T commented on gradlew:
```diff
@@ -133,26 +130,29 @@ location of your Java installation."
 else
     JAVACMD=java
-    which java >/dev/null 2>&1 || die "ERROR: JAVA_HOME is not set and no 'java' command could be found in your PATH.
+    if ! command -v java >/dev/null 2>&1
```
why you need to change this?

Lyam-T commented on gradle/wrapper/gradle-wrapper.properties:
```diff
@@ -2,5 +2,6 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 distributionUrl=https\://services.gradle.org/distributions/gradle-7.6.2-bin.zip
 networkTimeout=10000
+validateDistributionUrl=true
```
why you need to add this?

je-nnyt committed:
Merge branch 'master' into dev/11-add-delete-transportation

Need the parser function for deleteTransportation()

# Conflicts:
#	src/main/java/voyatrip/VoyaTrip.java
#	src/main/java/voyatrip/command/types/AccommodationCommand.java
#	src/main/java/voyatrip/command/types/Command.java
#	src/main/java/voyatrip/command/types/ItineraryCommand.java
#	src/main/java/voyatrip/command/types/TransportationCommand.java
#	src/main/java/voyatrip/command/types/TripsCommand.java
#	src/main/java/voyatrip/ui/Parser.java

je-nnyt committed:
Merge branch 'master' into dev/11-add-delete-transportation

je-nnyt committed:
Merge branch 'master' into dev/11-add-delete-transportation

# Conflicts:
#	build.gradle

je-nnyt committed:
Add Transportation class

je-nnyt committed:
Merge branch 'dev/11-add-delete-transportation' of https://github.com/je-nnyt/tp into dev/11-add-delete-transportation

# Conflicts:
#	src/main/java/voyatrip/Transportation.java

je-nnyt committed:
Fix merge conflicts from pulling from origin to dev/11-add-delete-transportation

je-nnyt committed:
Add Transportation class

je-nnyt committed:
Merge branch 'dev/11-add-delete-transportation' of https://github.com/je-nnyt/tp into dev/11-add-delete-transportation

# Conflicts:
#	src/main/java/voyatrip/Transportation.java

je-nnyt committed:
Modify Transportation constructor to have name as a parameter

je-nnyt committed:
Implement executeAddTransportation()
